### PR TITLE
NI-340 - consumer catalog item

### PR DESCRIPTION
**Pull request by WeiYewT**
### Background ###

- pass catalog item to the leaf nodes in the transformer
- dynamically layout stack in CatalogStackedCollectionComponent based on template row or column
- basic CatalogItem codable object

Fixes [([issue](https://rokt.atlassian.net/browse/NI-340))]

### What Has Changed: ###

List out what has changed as a result of this PR.

### How Has This Been Tested? ###

Describe the tests that you ran to verify your changes. 

### Notes

Add any notes or extra information here that might be useful to the reviewer if applicable i.e. links to documentation/blogs or dashboards.

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [x] All insignificant commits have been squashed.

### CODE REVIEW

File: Sources/RoktUXHelper/Services/LayoutTransformer/LayoutTransformer.swift [hunk 1]
```diff
@@ -298,7 +298,7 @@ struct LayoutTransformer<Expander: PayloadExpander, Extractor: DataExtractor> wh
     }
 
     func getDataImage(_ imageModel: DataImageModel<WhenPredicate>, slot: SlotOfferModel?) throws -> DataImageViewModel {
-        let creativeImage = slot?.offer?.creative.images?[imageModel.imageKey]
+        let creativeImage = slot?.offer?.creative.images?[imageModel.imageKey] ?? slot?.offer?.catalogItems?.first?.images?[imageModel.imageKey] // should have only 1 catalogItem at this node
```
Comment (danial-rokt) [inline Sources/RoktUXHelper/Services/LayoutTransformer/LayoutTransformer.swift:301]: Nit: No need to address it in this PR as you will work on the next part in a separate PR. I think we need to look into both creative image anc CatalogItem images based on the context as the name could exists on both images.

**Review by danial-rokt — APPROVED** [review]

File: Sources/RoktUXHelper/Data/Model/ExperienceResponse/CatalogItem.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+//  Licensed under the Rokt Software Development Kit (SDK) Terms of Use
+//  Version 2.0 (the "License");
+//
+//  You may not use this file except in compliance with the License.
+//
+//  You may obtain a copy of the License at https://rokt.com/sdk-license-2-0/
+
+import Foundation
+
+struct CatalogItem: Codable {
```
Comment (jamesnrokt) [inline Sources/RoktUXHelper/Data/Model/ExperienceResponse/CatalogItem.swift:14]: We probably need `ProviderData` serialized too
https://github.com/ROKT/cart-api/blob/6f4241c2dc1320b3ee3240d79a01bb882f1c8a7a/cart-api/src/Rokt.Cart.App.Interfaces/Pocos/CatalogItem.cs#L33C9-L33C104

Comment (danial-rokt) [reply Sources/RoktUXHelper/Data/Model/ExperienceResponse/CatalogItem.swift:14]: Yes I will add it in my PR when we support the partner event.

Comment (WeiYewT) [reply Sources/RoktUXHelper/Data/Model/ExperienceResponse/CatalogItem.swift:14]: Danial will be adding on to catalog items in his PR.

**Pull request merged**